File: kuaisupaixu/Sort.java
package kuaisupaixu;

public class Sort {
    public void quickSort(int[] arr) {
        if (arr == null || arr.length <= 1) {
            return;
        }
        quickSort(arr, 0, arr.length - 1);
    }
    
    private void quickSort(int[] arr, int left, int right) {
        if (left < right) {
            // 获取分区点
            int pivot = partition(arr, left, right);
            // 递归排序左半部分
            quickSort(arr, left, pivot - 1);
            // 递归排序右半部分
            quickSort(arr, pivot + 1, right);
        }
    }
    
    private int partition(int[] arr, int left, int right) {
        // 选择最右边的元素作为基准值
        int pivot = arr[right];
        int i = left - 1;
        
        // 将小于基准值的元素移到左边
        for (int j = left; j < right; j++) {
            if (arr[j] <= pivot) {
                i++;
                swap(arr, i, j);
            }
        }
        
        // 将基准值放到正确的位置
        swap(arr, i + 1, right);
        return i + 1;
    }
    
    private void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    // 测试代码
    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90};
        System.out.println("排序前：");
        printArray(arr);
        
        Sort sorter = new Sort();
        sorter.quickSort(arr);
        
        System.out.println("排序后：");
        printArray(arr);
    }
    
    private static void printArray(int[] arr) {
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}

File: kuaisupaixu/sort.py
def quick_sort(arr):
    if len(arr) <= 1:
        return arr
    
    # 选择基准值（这里选择第一个元素作为基准）
    pivot = arr[0]
    
    # 分区操作
    left = [x for x in arr[1:] if x <= pivot]    # 小于等于基准的元素
    right = [x for x in arr[1:] if x > pivot]    # 大于基准的元素
    
    # 递归排序左右两部分
    return quick_sort(left) + [pivot] + quick_sort(right)

# 使用示例
if __name__ == "__main__":
    # 测试数组
    test_array = [64, 34, 25, 12, 22, 11, 90]
    sorted_array = quick_sort(test_array)
    print("排序前:", test_array)
    print("排序后:", sorted_array)
